api/envoy/extensions/filters/http/on_demand/v3/on_demand.proto

```diff
@@ -11,11 +13,39 @@ option java_multiple_files = true;
 option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/on_demand/v3;on_demandv3";
 option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
-// [#protodoc-title: OnDemand]
-// IP tagging :ref:`configuration overview <config_http_filters_on_demand>`.
+// [#protodoc-title: On Demand Discovery]
+// On Demand Discovery :ref:`configuration overview <config_http_filters_on_demand>`.
 // [#extension: envoy.filters.http.on_demand]
 
+// On Demand Discovery filter config.
 message OnDemand {
```

Comment:
I am thinking about how we can merge ondemand message here for HCM here and for TcpProxy.

It turns out to be the message here has the role

1. registered message type for http.on_demand filter.
2. the ondemand CDS configuration, newly added in this PR.

Because of the first role, I think it's not appropriate to share this OnDemand message with tcp proxy

Reply:
The message for ODCDS config could be factored out and then reused here, so there could be still no duplication between on_demand and tcp_proxy extensions.